### PR TITLE
Deta drive put

### PR DESCRIPTION
**Pull request by Ninja243**

### CODE REVIEW

File: deta/base.py [hunk 1]
```diff
@@ -46,6 +50,12 @@ def append(self, value: typing.Union[dict, list, str, int, float, bool]):
 
     def prepend(self, value: typing.Union[dict, list, str, int, float, bool]):
         return self.Prepend(value)
+    
+    def measureSize(self, measurable) -> int:
+        if isinstance(measurable, BufferedIOBase):
+            return sys.getsizeof(measurable.read())
```
Comment (aavshr) [inline deta/base.py:56]: if we call read here already, are we not loading all of it into memory?

File: deta/base.py [hunk 2]
```diff
@@ -46,6 +50,12 @@ def append(self, value: typing.Union[dict, list, str, int, float, bool]):
 
     def prepend(self, value: typing.Union[dict, list, str, int, float, bool]):
         return self.Prepend(value)
+    
+    def measureSize(self, measurable) -> int:
```
Comment (aavshr) [inline deta/base.py]: should be `snake_case` `measure_size`

File: deta/base.py [hunk 3]
```diff
@@ -100,11 +122,47 @@ def __init__(self, drive_name:str=None, project_key:str=None, project_id:str=Non
...
+        if not isinstance(data, BufferedIOBase):
+            _, res = self._request(f"/files?name={name}", "POST", data, content_type)
+            return res["name"]
+        retry_queue = []
+        def _partial_upload():
+            chunk_number = 1
+            uuid = ""
+            for chunk in iter(partial(data.read, chunk_size), b''):
+                if (chunk_number == 1) and (self.util.measureSize(chunk) < chunk_size):
+                    print(chunk)
```
Comment (aavshr) [inline deta/base.py]: remove `print`

File: deta/base.py (hunk 2 above)
Comment (aavshr) [reply deta/base.py]: shouldn't be in `Util` class, it's only for base

File: deta/base.py [hunk 4]
```diff
@@ -100,11 +122,47 @@ def __init__(self, drive_name:str=None, project_key:str=None, project_id:str=Non
...
+                    except Exception as e:
+                        print(f"[!] Chunk {chunk_number} of size {chunk_size} failed to upload. Added to retry queue.")
+                        retry_queue.append(chunk_number)
+                    chunk_number = chunk_number + 1
+                    _, res = self._request(f"/uploads/{uuid}?name={name}", "PATCH")
+                    return res["name"]
+
+        with data:
+            _partial_upload()
+            while len(retry_queue) > 0:
```
Comment (aavshr) [inline deta/base.py]: chance of executing infinitely if a chunk upload fails as it's not removed from the `retry_queue` if upload is successfull, we shouldn't worry about retries for now

File: deta/base.py (hunk 2 above)
Comment (Ninja243) [reply deta/base.py]: Gotcha, thanks!

**Pull request merged**